pkg/local_workflows/ignore_workflow/ignore_workflow.go

```diff
@@ -0,0 +1,333 @@
+	expireDisplayText := "Does not expire"
+	if expire != nil {
+		expireDisplayText = expire.Format(time.DateOnly)
+	}
+	return fmt.Sprintf("  Requested on:  %s\n  Requested by:  %s\n  Expiration:    %s\n  Ignore type:   %s", requestedOn, userName, expireDisplayText, ignoreType)
+}
+
+func getExpireValue(config configuration.Configuration) (*time.Time, error) {
+	shouldParse := config.IsSet(ExpirationKey) || config.GetBool(InteractiveKey)
+	if !shouldParse {
+		//nolint:nilnil // returning nil,nil here means that there is no expiration, and we didn't run into an error which is a valid case
+		return nil, nil
+	}
+	expireStr, err := config.GetStringWithError(ExpirationKey)
+	if err != nil || expireStr == "" {
+		return nil, err
+	}
+
+	expireVal, parseErr := time.Parse(time.DateOnly, expireStr)
+	if parseErr != nil {
+		return nil, err
+	}
+
+	return &expireVal, nil
+}
+
+func createIgnoreWorkflowData(id workflow.Identifier, config configuration.Configuration, ignoreResponse *sarif.Suppression, logger *zerolog.Logger) (workflow.Data, error) {
+	if ignoreResponse == nil {
+		return nil, fmt.Errorf("ignore response is nil")
+	}
+	marshalledObj, err := json.Marshal(ignoreResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	data := workflow.NewData(
+		id,
+		"application/json",
+		marshalledObj,
+		workflow.WithConfiguration(config),
+		workflow.WithLogger(logger),
+	)
+
+	return data, nil
+}
+
+func createPayload(repoUrl string, branchName string, expire *time.Time, ignoreType string, reason string, findingsId string) policyApi.CreatePolicyPayload {
+	meta := policyApi.Meta{}
+	meta["repo_url"] = repoUrl
+	meta["branch_name"] = branchName
+
+	var payload policyApi.CreateOrgPolicyApplicationVndAPIPlusJSONRequestBody
+	payload.Data.Meta = &meta
+	payload.Data.Attributes.Action.Data = policyApi.PolicyActionIgnoreData{
+		Expires:    expire,
+		IgnoreType: policyApi.PolicyActionIgnoreDataIgnoreType(ignoreType),
+		Reason:     &reason,
+	}
+
+	payload.Data.Attributes.ActionType = policyApi.PolicyAttributesActionTypeIgnore
+	payload.Data.Attributes.ConditionsGroup = policyApi.PolicyConditionsGroup{
+		Conditions: []policyApi.PolicyCondition{
+			{
+				Field:    policyApi.Snykassetfindingv1,
+				Operator: policyApi.Includes,
+				Value:    findingsId,
+			},
+		},
+		LogicalOperator: policyApi.And,
+	}
+	payload.Data.Type = policyApi.CreatePolicyPayloadDataTypePolicy
+	return payload
+}
+
+func getUser(invocationCtx workflow.InvocationContext) (string, error) {
+	config := invocationCtx.GetConfiguration().Clone()
+	config.Set(configuration.FLAG_EXPERIMENTAL, true)
+
+	data, err := invocationCtx.GetEngine().InvokeWithConfig(localworkflows.WORKFLOWID_WHOAMI, config)
+	if err != nil {
+		return "", err
+	}
+
+	noAuthErr := fmt.Errorf("user is not authenticated")
+	if len(data) > 0 {
+		user, ok := data[0].GetPayload().(string)
+		if !ok {
+			return "", noAuthErr
+		}
+		return user, nil
+	} else {
+		return "", noAuthErr
+	}
```

Comment:
question: is this necessary? the relevant error should bubble up from the invocation, this extra logic seems to just obfusicate the original error